src/algebra/gcd_monoid/basic.lean

```diff
@@ -208,44 +211,53 @@ end associates
 supremum, `1` is bottom, and `0` is top. The type class focuses on `gcd` and we derive the
```

Comment:
Doesn't a bounded lattice require `lcm` and `gcd` be commutative and associative, which is no longer true of the weaker typeclass?

Reply:
That seems plausible - I can't find an actual Lean instance, though, so I'm not sure what this comment is referring to.

Reply:
Do the generalisations you care about not have associativity or commutativity either?

Reply:
No commutativity, in any case:
```lean
import algebra.euclidean_domain

#eval euclidean_domain.gcd (1 : int) (-1 : int)
-- 1
#eval euclidean_domain.gcd (-1 : int) (1 : int)
-- -1
```